sdk/core/core-lro/src/poller.ts

```diff
@@ -0,0 +1,138 @@
+import { PollOperation } from "./pollOperation";
+
+export type CancelOnProgress = () => void;
+
+export type PollProgressCallback<TProperties> = (properties: TProperties) => void;
+
+export class PollerStoppedError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "PollerStoppedError";
+    Object.setPrototypeOf(this, PollerStoppedError.prototype);
+  }
+}
+
+export abstract class Poller<TProperties, TResult> {
+  private stopped: boolean;
+  private resolve?: (value?: TResult) => void;
+  private reject?: (error: PollerStoppedError | Error) => void;
+  private pollOncePromise?: Promise<void>;
+  private cancelPromise?: Promise<PollOperation<TProperties, TResult>>;
+  private promise: Promise<TResult>;
+  private pollProgressCallbacks: PollProgressCallback<TProperties>[] = [];
+  public operation: PollOperation<TProperties, TResult>;
```

Comment:
let's make `operation` private, since I don't think we want to expose `PollOperation`'s methods to the client. Instead let's add two new functions `getState()` and `getProperties()` that return `PollOperationState<TResult>` and `TProperties` respectively.